TCC.Lib/TarCompressCrypt.cs

```diff
@@ -198,12 +226,25 @@ private static string CompressCommand(Block block, TccOption option, ExternalDep
 
         private static string DecompressCommand(Block block, TccOption option, ExternalDependecies ext)
         {
-            string cmd;
+            string cmd = null;
             switch (option.PasswordOption.PasswordMode)
             {
                 case PasswordMode.None:
                     //lz4 archive.tar.lz4 -dc --no-sparse | tar xf -
-                    cmd = $"{ext.Lz4().Escape()} {block.Source.Escape()} -dc --no-sparse ";
+                    switch (option.Algo)
+                    {
+                        case CompressionAlgo.Lz4:
+                            cmd = $"{ext.Lz4().Escape()} {block.Source.Escape()} -dc --no-sparse ";
+                            break;
+                        case CompressionAlgo.Brotli:
+                            cmd = $"{ext.Brotli().Escape()} {block.Source.Escape()} -d -c ";
+                            break;
+                        case CompressionAlgo.Zstd:
+                            cmd = $"{ext.Zstd().Escape()} {block.Source.Escape()} -d -c ";
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException("Unknown PasswordMode");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use a constructor overloads that allows a more meaningful exception message to be provided. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3928)